File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/domene/InnloggetBruker.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.domene;

import lombok.Data;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.altinn.AltinnOrganisasjon;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.tilgangskontroll.TilgangskontrollException;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

@Data
public class InnloggetBruker {
    private List<AltinnOrganisasjon> organisasjoner;
    private Fnr fnr;

    public InnloggetBruker(Fnr fnr) {
        this.fnr = fnr;
        organisasjoner = new ArrayList<>();
    }

    public void sjekkTilgang(Orgnr orgnr) {
        if (!harTilgang(orgnr)) {
            throw new TilgangskontrollException("Har ikke tilgang til statistikk for denne bedriften.");
        }
    }

    public boolean harTilgang(Orgnr orgnr) {
        List<String> orgnumreBrukerHarTilgangTil = organisasjoner
                .stream()
                .filter(Objects::nonNull)
                .map(org -> org.getOrganizationNumber())
                .collect(Collectors.toList());

        return orgnumreBrukerHarTilgangTil.contains(orgnr.getVerdi());
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/tilgangsstyring/InnloggetBruker.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.tilgangsstyring

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.fellesdomene.Orgnr
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.altinn.AltinnOrganisasjon
import java.util.*
import java.util.stream.Collectors

data class InnloggetBruker(
    val fnr: Fnr,
    var brukerensOrganisasjoner: List<AltinnOrganisasjon> = emptyList()
) {

    fun harTilgang(orgnr: Orgnr): Boolean {
        val orgnumreBrukerHarTilgangTil = brukerensOrganisasjoner.stream()
            .filter { obj: AltinnOrganisasjon? -> Objects.nonNull(obj) }
            .map(AltinnOrganisasjon::organizationNumber)
            .collect(Collectors.toList())
        return orgnumreBrukerHarTilgangTil.contains(orgnr.verdi)
    }
}
